### PR TITLE
OPAL-626 (Added clear_index function with optional refresh param)

### PR DESCRIPTION
**Pull request by JA-USAF**
Adds a function which clears a given index. An optional refresh parameter can be used to re-generate the index after clearing.

Also adds function to drop the sql files/schemas (for sqlite/sql). drop_index() only exists in SQL/SQLite implementations, and will drop/delete the schemas/file. This is useful when using a demo pantry and you need to clean up the schema afterwards.

Note: The pandas clear_index() function behaves slightly differently than the SQL/SQLite due to the implementation of the pandas index backend.

### CODE REVIEW

File: weave/index/index_sqlite.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ def generate_index(self, **kwargs):
 
         Parameters
         ----------
+        refresh: bool (default: False)
```
Comment (AM-USAF) [inline weave/index/index_sqlite.py:98]: Was this supposed to be added to the generate_index docstring?

**Review by JN-USAF — DISMISSED** [review]
I tested the clear_index function out in a jupyter notebook, and it's doing what I expect.

File: weave/index/index_sqlite.py (hunk 1 above)
Comment (JA-USAF) [reply weave/index/index_sqlite.py]: Good catch--removed 👍

**Review by JS-USAF — APPROVED** [review]
Looks good!

**Review by AM-USAF — APPROVED** [review]

**Pull request merged**